swarm/pss/prox_test.go

```diff
@@ -133,99 +123,132 @@ func readSnapshot(t *testing.T, nodeCount int) simulations.Snapshot {
 	}
-	log.Debug("msgs to receive", "count", msgsToReceive)
+	log.Debug("msgs to receive", "count", d.requiredMessages)
 }
 
 func TestProxNetwork(t *testing.T) {
-	if (*runNodes > 0 && *runMessages == 0) || (*runMessages > 0 && *runNodes == 0) {
-		t.Fatal("cannot specify only one of flags --nodes and --messages")
-	} else if *runNodes > 0 {
-		t.Run(fmt.Sprintf("%d/%d", *runMessages, *runNodes), testProxNetwork)
-	} else {
-		t.Run("1/4", testProxNetwork)
+	t.Run("16/16", testProxNetwork)
+}
+
+// params in run name: nodes/msgs
+func TestProxNetworkLong(t *testing.T) {
+	if !*longrunning {
+		t.Skip("run with --longrunning flag to run extensive network tests")
 	}
+	t.Run("8/100", testProxNetwork)
+	t.Run("16/100", testProxNetwork)
+	t.Run("32/100", testProxNetwork)
+	t.Run("64/100", testProxNetwork)
 }
 
-// This tests generates a sequenced number of messages with random addresses.
-// It then calculates which nodes in the network have the address of each message
-// within their nearest neighborhood depth, and stores them as recipients.
-// Upon sending the messages, it verifies that the respective message is passed to the message handlers of these recipients.
-// It will fail if a recipient handles a message it should not, or if after propagation not all expected messages are handled (timeout)
+// This tests generates a number of messages with random addresses. Then,
+// for each message it calculates which nodes in the network the msg address
+// within its nearest neighborhood depth, and stores those nodes as possible
+// recipients. Those nodes that are the closest to the message address (nodes
+// belonging to the deepest PO wrt msg address) are stored as required recipients.
+// Upon sending the messages, the test verifies that the respective message is
+// passed to the message handlers of these required recipients. Test will fail
+// if a message is handled by recipient which is not listed among the allowed
+// recipients of this particular message. It also fails after timeout, if not
+// all the required recipients have received their respective messages.
+//
+// For example, if proximity order of certain msg address is 4, and node X
+// has PO=5 wrt the message address, and nodes Y and Z have PO=6, then:
+// nodes Y and Z will be considered required recipients of the msg,
+// whereas nodes X, Y and Z will be allowed recipients.
 func testProxNetwork(t *testing.T) {
-	resetTestVariables()
+	var tstdata testData
 	msgCount, nodeCount := getCmdParams(t)
 	handlerContextFuncs := make(map[Topic]handlerContextFunc)
 	handlerContextFuncs[topic] = nodeMsgHandler
-	services := newProxServices(true, handlerContextFuncs, kademlias)
-	snap := readSnapshot(t, nodeCount)
-	sim = simulation.New(services)
-	defer sim.Close()
-	err := sim.Net.Load(&snap)
+	tstdata.kademlias = make(map[enode.ID]*network.Kademlia)
```

Comment:
See above, would also go in init method.

Reply:
it can not, because we need kademlias before we call init

Reply:
I don't understand this. I suggested that tstdata should have a constructor, which takes care of all allocations (line 128-137) and then could take care of this allocation, too. Why isn't that possible? Or are you against the suggestion? (since you didn't change it, iether)